Lua/Lua.sublime-syntax

```diff
@@ -680,6 +698,38 @@ contexts:
     - match: \\.
       scope: invalid.illegal.invalid-escape.lua
 
+  # pop the context once one escaped character has been matched
+  string-escape-pop:
+    - match: \\[abfnrtv\\'"\[\]\n]
+      scope: constant.character.escape.lua
+      pop: 1
+
+    - match: \\x\h\h
+      scope: constant.character.escape.hexadecimal.lua
+      pop: 1
+
+    - match: \\\d{1,3}
+      scope: constant.character.escape.decimal.lua
+      pop: 1
+
+    # unicode can make multiple, but that is a bit hard to handle
+    - match: \\u\{\h+\}
+      scope: constant.character.escape.unicode.lua
+      pop: 1
+
+    # \z turns into nothing, so we don't pop
+    - match: \\z
+      push:
+        - meta_scope: constant.character.escape.whitespace.lua
+        - include: else-pop
```

Comment:
Looks like just consumming `\\z` without pushing or popping contexts is enough

```suggestion
    - match: \\z
      scope: constant.character.escape.whitespace.lua
```

Reply:
that won't be enough, since it has to consume whitespace, including newlines. and newlines will get caught on the next match if this doesn't have the extra context